src/Common/src/Interop/OSX/Interop.libproc.cs

```diff
@@ -0,0 +1,523 @@
+            if (result <= 0)
+            {
+                throw new System.ComponentModel.Win32Exception();
+            }
+
+            List<proc_fdinfo> fds = new List<proc_fdinfo>();
+
+            // Copy the file descriptors over
+            for (int i = 0; i < (result / PROC_PIDLISTFD_SIZE); i++)
+            {
+                fds.Add(Marshal.PtrToStructure<proc_fdinfo>(buffer + (i * PROC_PIDLISTFD_SIZE)));
+            }
+
+            // Free our unmanaged memory
+            Marshal.FreeCoTaskMem(buffer);
+
+            return fds;
+        }
+
+        private static bool internal_proc_pidinfo_helper<T>(int pid, int flavor, ulong args, out T info)
+        {
+            // Negative PIDs are invalid
+            if (pid < 0)
+            {
+                throw new ArgumentOutOfRangeException("pid", SR.NegativePidNotSupported);
+            }
+
+            bool success = false;
+
+            // Allocate our unmanaged buffer for the task info
+            info = default(T);
+            int size = Marshal.SizeOf<T>();
+            IntPtr buffer = Marshal.AllocCoTaskMem(size);
+
+            // Get the process information for the given PID
+            int result = proc_pidinfo(pid, flavor, args, buffer, size);
+            if (result == size)
+            {
+                // Copy the struct from unmanaged memory to managed memory
+                info = Marshal.PtrToStructure<T>(buffer);
+                success = true;
+            }
+
+            // Free our unmanaged buffer
+            Marshal.FreeCoTaskMem(buffer);
+
+            return success;
+        }
+
+        /// <summary>
+        /// Gets the full path to the executable file identified by the specified PID
+        /// </summary>
+        /// <param name="pid">The PID of the running process</param>
+        /// <param name="buffer">A pointer to an allocated block of memory that will be filled with the process path</param>
+        /// <param name="bufferSize">The size of the buffer, should be PROC_PIDPATHINFO_MAXSIZE</param>
+        /// <returns>Returns the length of the path returned on success</returns>
+        [DllImport(Interop.Libraries.libproc, CharSet = CharSet.Ansi, SetLastError = true)]
+        private static extern int proc_pidpath(
+            int pid, 
+            IntPtr buffer, 
+            uint bufferSize);
+
+        /// <summary>
+        /// Gets the full path to the executable file identified by the specified PID
+        /// </summary>
+        /// <param name="pid">The PID of the running process</param>
+        /// <returns>Returns the full path to the process executable</returns>
+        internal static string proc_pidpath(int pid)
+        {
+            // Negative PIDs are invalid
+            if (pid < 0)
+            {
+                throw new ArgumentOutOfRangeException("pid", SR.NegativePidNotSupported);
+            }
+
+            // Allocate our unmanaged buffer for the path
+            string path = string.Empty;
```

Comment:
Nit: why does this need to be defined and initialized here?  Seems like you could just delete this line and add a variable type to the line later that sets path to the result of the Marshal call.